components/hook-cjs/void/index.test.mjs

```diff
@@ -12,7 +12,7 @@ assertDeepEqual(
   await testHookAsync(
     component,
     { configuration: { hooks: { cjs: false } } },
-    async (state) => null,
+    async (_state) => null,
```

Comment:
`_state` can be omitted if it's unused. `async` is also not required because `await null` will still resolve to `null`.

Reply:
I intentionally leave unused arguments for documentation purpose. Also, I try to respect function arity. I know this not required in js but it is a habit I carried from using other languages. The codebase is full of those. Is it something we can live with?

Reply:
It is the same idea for tagging functions that will be used as asynchronous with `async`. I find it abhorrent that `await` works for pure value. Again this is a habit that I carry from other languages.

Reply:
Yeah I know that it adds a micro task scheduling and that there is no objective reason to keep these `async` aside from satisfying my OCD. `require-await` enabled in 0a6e3f6e9bc8deb9262787b90cbbc424b75b0bd1

Reply:
@dividedmind I don't like the design decision of js to "recover" from errors. For instance: the entire builtin library that performs type casting and function arity mismatch. IMO, pretty much any other language that is strongly typed is better in that regard (I have the most coding experience with Scheme and Haskell).

Reply:
You really should start using TypeScript, you're going to love it :) There are ways to incrementally migrate a codebase to it if you're interested: https://www.typescriptlang.org/docs/handbook/migrating-from-javascript.html

To be fair, JS is much less lenient (in a good way) nowadays (in strict mode).